### PR TITLE
[TechDocs] Page layout improvements

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Some page layout tweaks:

- [x] drawer toggle margins
- [x] code block margins
- [x] sidebar drawer width
- [x] inner content width
- [x] footer link width
- [x] sidebar table of contents scroll

| Before | After |
| :-----: | :----: |
|  mobile content alignment (old) | mobile content alignment |
| mobile sidebar width (old) | mobile sidebar width |

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs/src/reader/components/Reader.tsx [hunk 1]
```diff
@@ -556,20 +556,30 @@ export const useTechDocsReaderDom = (
                 height: 100%;
               }
               .md-sidebar--primary {
-                width: 12.1rem !important;
+                width: 16rem !important;
                 z-index: 200;
                 left: ${
-                  isPinned ? 'calc(-12.1rem + 242px)' : 'calc(-12.1rem + 72px)'
+                  isPinned ? 'calc(-16rem + 242px)' : 'calc(-16rem + 72px)'
```
Comment (realandersn) [inline plugins/techdocs/src/reader/components/Reader.tsx:562]: Looks wrong when sidebar is pinned.

```suggestion
                  isPinned ? 'calc(-16rem + 224px)' : 'calc(-16rem + 72px)'
```

**Review by realandersn — CHANGES_REQUESTED** [review]
Nice! Just found one thing that would be good to fix.

File: plugins/techdocs/src/reader/components/Reader.tsx (hunk 1 above)
Comment (camilaibs) [reply plugins/techdocs/src/reader/components/Reader.tsx]: Hey, good catch!
I pushed a new commit using the sidebarConfig instead, can you review it again, please?

File: plugins/techdocs/src/reader/components/Reader.tsx [hunk 2]
```diff
@@ -585,16 +600,16 @@ export const useTechDocsReaderDom = (
...
               }
             }
 
             @media screen and (max-width: 600px) {
               .md-sidebar--primary {
-                left: -12.1rem !important;
-                width: 12.1rem;
+                left: -16rem !important;
+                width: 16rem;
               }
```
Comment (realandersn) [inline plugins/techdocs/src/reader/components/Reader.tsx:613]: Found an issue with the nav on mobile: it gets cut off by the Backstage side nav (which is at the bottom on mobile view). Including the following style here seems to fix it:

```
.md-sidebar--primary .md-sidebar__scrollwrap {
  bottom: ${sidebarConfig.mobileSidebarHeight}px;
}
```

Comment (camilaibs) [reply plugins/techdocs/src/reader/components/Reader.tsx:616]: Thanks again, fixed 🙌🏻

**Review by realandersn — APPROVED** [review]
:shipit:

**Pull request merged**